src/lib/optimizer/strategy/abstract_rule.cpp

```diff
@@ -9,9 +9,33 @@
 
 namespace opossum {
 
-void AbstractRule::_apply_to_inputs(std::shared_ptr<AbstractLQPNode> node) const {  // NOLINT
-  if (node->left_input()) apply_to(node->left_input());
-  if (node->right_input()) apply_to(node->right_input());
+void AbstractRule::apply_to_plan(const std::shared_ptr<LogicalPlanRootNode>& lqp_root) const {
+  // (1) Optimize root LQP
+  _apply_to_plan_without_subqueries(lqp_root);
+
+  // (2) Optimize distinct subquery LQPs, one-by-one.
+  auto subquery_expressions_by_lqp = collect_subquery_expressions_by_lqp(lqp_root);
+  for (const auto& [lqp, subquery_expressions] : subquery_expressions_by_lqp) {
+    if (std::all_of(subquery_expressions.cbegin(), subquery_expressions.cend(),
+                    [](auto subquery_expression) { return subquery_expression.expired(); })) {
+      continue;
+    }
+    // (2.1) Prepare
```

Comment:
Not sure what this comment contributes. Also, could you add some newlines beteween the blocks?

Reply:
True, removed.